add intercomparison metrics calculator and tc for qa4sm

wpreimes opened the pull request.

wpreimes commented:
@tracyscanlon maybe before merging this, lets discuss it with the others, also I might add another commit here.

tracyscanlon commented:
@wpreimes Let's discuss with the others. And yes, it will need testing.

sebhahn commented on pytesmo/validation_framework/metric_calculators.py:
```diff
@@ -236,3 +240,725 @@ def calc_metrics(self, data, gpi_info):
+
+
+
+if __name__  == '__main__':
```
this can be removed

wpreimes replied:
ok, thanks, @sebhahn have you noticed that here the wrong R is used? https://github.com/wpreimes/pytesmo/blob/d715209417c6165f599d7e9848b7ca10b52a2b19/pytesmo/validation_framework/metric_calculators.py#L482

sebhahn replied:
No, could you fix this? Is not a big problem though, because we do not use spearman

wpreimes replied:
sure, already fixed it

wpreimes commented:
I have also removed the old ISMN code and added the ISMN reader package as a dependecy (same as in #124 , which can be closed after merging this as well) and updated the tests accordingly

sebhahn commented:
Is this pr ready?

wpreimes commented:
Yes you can merge it from my side

tracyscanlon commented:
No objections here. Please go ahead and merge.


From: Wolfgang Preimesberger <notifications@github.com>
Sent: Dienstag, 09. April 2019 11:49
To: TUW-GEO/pytesmo <pytesmo@noreply.github.com>
Cc: Scanlon, Tracy Marie <tracy.scanlon@geo.tuwien.ac.at>; Mention <mention@noreply.github.com>
Subject: Re: [TUW-GEO/pytesmo] add intercomparison metrics calculator and tc for qa4sm (#156)


Yes you can merge it from my side

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/TUW-GEO/pytesmo/pull/156#issuecomment-481182614>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AgTLFwS3vTcH1xOZJxb3QcbSXMCXoyHCks5vfGH2gaJpZM4brX5N>.

cpaulik commented:
Two small comments:

I don't think we should add `ascat` and `ismn` as requirements of pytesmo. This somewhat defeats the purpose of splitting them into separate packages if they are not installable separately. We could put them into the `extras` section of `setup.cfg` then somebody could e.g. install `pip install pytesmo[ascat,ismn]` and get all dependencies that are required to work with those datasources. We could also make an extra `dataset-readers` which could include `ascat,gldas,ismn,ecmwf...` so installing `pip install pytesmo[dataset-readers]` would install all of them.

The second comment/question is about the `build_filename` method that limits the filename to 144 characters. Where did the 144 come from? On Linux the filename has a limit of 255 bytes so there it should not be a problem. On Windows the whole path can not be longer than 255 characters so there we might even have a problem with 144 if the basepath is too long.

wpreimes commented:
Good ideas. We will look into this! Yes the filename function needs some rework, the 144 characters came from my windows machine (as win counts the whole path as you said). I will think of a better solution.
